app/main.py

```diff
@@ -1,13 +1,22 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": f"{file_path}",
+            "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [{"errors": [format_linter_error(error) for error in value],
+             "path": f"{key}",
+             "status": "passed" if linter_report[key] == [] else "failed"}
+            for key, value in linter_report.items()]
```

Comment:
everything looks good now but use descriptive and correct variable names(instead of `key` and `value` you can use file_path and errors or something like that)